src/main/java/com/github/tomakehurst/wiremock/store/InMemoryObjectStore.java

```diff
@@ -54,27 +60,39 @@ public Stream<String> getAllKeys() {
 
   @Override
   public void put(String key, Object content) {
-    cache.put(key, content);
+    Object previousValue = cache.put(key, content);
     touchAndResize(key);
+    handleEvent(new StoreEvent<>(key, previousValue, content));
   }
 
   @Override
   @SuppressWarnings("unchecked")
   public <T> T compute(String key, Function<T, T> valueFunction) {
+    final AtomicReference<T> previousValue = new AtomicReference<>();
     final T result =
-        (T) cache.compute(key, (k, currentValue) -> valueFunction.apply((T) currentValue));
+        (T)
+            cache.compute(
+                key,
+                (k, currentValue) -> {
+                  previousValue.set((T) currentValue);
+                  return valueFunction.apply((T) currentValue);
+                });
     if (result != null) {
       touchAndResize(key);
     } else {
       keyUseOrder.remove(key);
     }
+    handleEvent(new StoreEvent<>(key, previousValue.get(), result));
```

Comment:
Might be nicer to have two different static factory methods i.e.
`StoreEvent.set(key, prevValue, newValue()` and `StoreEvent.remove(key, prevValue)`

Reply:
yeah that's nice

Reply:
done